__tests__/app.test.js

```diff
@@ -41,28 +46,109 @@ describe("GET/api/categories requests.", () => {
+        return request(app).patch('/api/reviews/2')
+        .expect(200)
+        .send({ inc_votes: 5 })
+        .then(({body}) => {
+            expect(body.review.votes).toBe(10);
+        })
+    })
+
 })
 
-describe("Error handling.", () => {
-    test("Responds to bad paths with 404 status and relevant message.", () => {
+describe("Errors: Bad paths.", () => {
+
+    test("Responds to bad categories path with 404 status and relevant message.", () => {
         return request(app).get('/api/category')
         .expect(404)
         .then(({body}) => {
             expect(body.message).toBe("That page does not exist.");
         })
     })
-    test("Responds to valid-but-non-existent paths with 404 status and relevant message.", () => {
+
+    test("Responds to bad reviews path with 404 status and relevant message.", () => {
+        return request(app).get('/api/reveiws/2')
+        .expect(404)
+        .then(({body}) => {
+            expect(body.message).toBe("That page does not exist.");
+        })
+    })
+
+})
+
+describe("Errors: Valid but non-existent paths.", () => {
+
+    test("GET request: Responds to valid but non-existent Review ID with 404 status and relevant message.", () => {
         return request(app).get('/api/reviews/1000')
         .expect(404)
         .then(({body}) => {
             expect(body.message).toBe("That Review ID doesn't exist.");
         })
     })
-    test("Responds to invalid paths with 404 status and relevant message.", () => {
+
+    test("PATCH request: Responds to valid but non-existent Review ID with 404 status and relevant message.", () => {
+        return request(app).patch('/api/reviews/1000')
+        .expect(404)
+        .send({ inc_votes: 5 })
+        .then(({body}) => {
+            expect(body.message).toBe("That Review ID doesn't exist.");
+        })
+    })
+
+})
+
+describe("Errors: Invalid paths/ input objects.", () => {
+
+    test("Responds to invalid Review ID with 400 status and relevant message.", () => {
         return request(app).get('/api/reviews/apple')
         .expect(400)
         .then(({body}) => {
-            expect(body.message).toBe("Invalid Review ID.");
+            expect(body.message).toBe("Invalid Review ID or input object.");
+        })
+    })
+
+    test("Responds to invalid input on patch request with 400 status and relevant message.", () => {
+        return request(app).patch('/api/reviews/2')
+        .expect(400)
+        .send({ inc_votes: 'A' })
+        .then(({body}) => {
+            expect(body.message).toBe("Invalid Review ID or input object.");
         })
     })
+
+    test("Responds to empty object on patch request with 400 status and relevant message.", () => {
+        return request(app).patch('/api/reviews/2')
+        .expect(400)
+        .send({})
+        .then(({body}) => {
+            expect(body.message).toBe("Invalid input object.");
+        })
+    })
+
```

Comment:
What happens if we try to patch a url where the id is the incorrect data type?

Reply:
Thanks Emily. I've restructured the test suites now. If the user uses an invalid review id for a patch request they should now get a 400 status and a 'Invalid Review ID' message.